sim/paladin/items.go

```diff
@@ -227,3 +234,51 @@ func crusadersZealAura465414(character *core.Character) *core.Aura {
 		},
 	})
 }
+
+func libramOfWrathAura(agent core.Agent) {
+	paladin := agent.(PaladinAgent).GetPaladin()
+	holyWrathSpells := []*core.Spell{}
+
+	buffAura := paladin.RegisterAura(core.Aura{
+		ActionID:  core.ActionID{SpellID: 470246},
+		Label:     "Libram Of Wrath Buff",
+		Duration:  time.Second * 15,
+		MaxStacks: 5,
+		OnInit: func(aura *core.Aura, sim *core.Simulation) {
+			holyWrathSpells = core.FilterSlice(paladin.holyWrath, func(spell *core.Spell) bool { return spell != nil })
+		},
+		OnStacksChange: func(aura *core.Aura, sim *core.Simulation, oldStacks int32, newStacks int32) {
+			core.Each(holyWrathSpells, func(spell *core.Spell) {
+				spell.CastTimeMultiplier += (0.2*float64(oldStacks))
+				spell.Cost.Multiplier += int32(100.0 * (0.2*float64(oldStacks)))
+				spell.DamageMultiplier -= (0.2*float64(oldStacks))
+
+				spell.CastTimeMultiplier -= (0.2*float64(newStacks))
+				spell.Cost.Multiplier -= int32(100.0 * (0.2*float64(newStacks)))
+				spell.DamageMultiplier += (0.2*float64(newStacks))
+				
+
+			})
+		},
+		OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {			
+			if spell.SpellCode == SpellCode_PaladinHolyWrath {
+				aura.Deactivate(sim)
+			}
+		},
+	})
+
+	paladin.RegisterAura(core.Aura{
+		Label:    "Libram Of Wrath Trigger",
+		Duration: core.NeverExpires,
+		OnReset: func(aura *core.Aura, sim *core.Simulation) {
+			
+			aura.Activate(sim)
+		},
+		OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if spell.SpellCode == SpellCode_PaladinHolyShock {
```

Comment:
Does the shock need to hit to proc the libram or can a full resist still proc?

Reply:
went and tested, it does proc on full resist - I changed it to OnCastComplete is that the appropriate trigger with that?

Reply:
Then yeah it's probably an on cast complete!